src/simularium/ISimulator.ts

```diff
@@ -15,46 +22,27 @@ export interface ISimulator {
         handler: (msg: VisDataMessage | ArrayBuffer) => void
     ): void;
 
-    /**
-     * Connect
-     * */
-    connectToRemoteServer(address: string): Promise<string>;
-    socketIsValid(): boolean;
-    isConnectedToRemoteServer(): boolean;
-    getIp(): string;
-    disconnect(): void;
+    setErrorHandler(handler: (msg: Error) => void): void;
 
-    // TODO Fold into sendUpdate
-    sendTimeStepUpdate(newTimeStep: number): void;
-    sendUpdate(data: Record<string, unknown>): void;
+    // general update function for client simulators
+    // todo: could also be implemented for control of live
+    // and pre-run remote simulations
+    sendUpdate(data: Record<string, unknown>): Promise<void>;
 
     /**
-     * Simulation Control
+     * General Simulation Control
      *
-     * Simulation Run Modes:
-     *  Live : Results are sent as they are calculated
-     *  Pre-Run : All results are evaluated, then sent piecemeal
-     *  Trajectory File: No simulation run, stream a result file piecemeal
+     * initialize: prepare a simulation for streaming, async to accommodate connection steps
+     * destroy: stop sending frames, release resources, close connections
+     * stream: request a stream of frames
+     * pause: request that frames stop being sent, keep the precomputed streaming head where it is
```

Comment:
I suggest moving these to above each of the individual functions.

Reply:
Agreed!